internal/app/fipcontroller/kubernetes.go

```diff
@@ -82,11 +95,7 @@ func (controller *Controller) nodeAddressList(ctx context.Context, nodeAddressTy
 		}
 		controller.Logger.Debugf("Using address type '%s' for node %s", checkAddressType, node.Name)
 
-		address := searchForAddress(addresses, checkAddressType)
-		if address == nil {
-			return nil, fmt.Errorf("coud not find address for node %s", node.Name)
-		}
-		addressList = append(addressList, address)
+		addressList = append(addressList, searchForAddresses(addresses))
 	}
 
 	if len(addressList) < 1 {
```

Comment:
`note` I think `len(addressList) == 0` is a more natural way to check for an empty slice